burn-core/src/nn/loss/cross_entropy.rs

```diff
@@ -25,36 +111,128 @@ impl<B: Backend> CrossEntropyLoss<B> {
     /// - logits: `[batch_size, num_targets]`
     /// - targets: `[batch_size]`
     pub fn forward(&self, logits: Tensor<B, 2>, targets: Tensor<B, 1, Int>) -> Tensor<B, 1> {
+        match (self.smoothing, self.binary) {
+            (Some(_), true) => self.forward_binary(logits, targets),
+            (Some(alpha), false) => self.forward_smoothed(logits, targets, alpha),
+            _ => self.forward_default(logits, targets),
+        }
+    }
+
+    fn forward_binary(&self, logits: Tensor<B, 2>, mut targets: Tensor<B, 1, Int>) -> Tensor<B, 1> {
+        let [batch_size, nr_classes] = logits.dims();
+
+        if let Some(alpha) = self.smoothing {
+            targets = targets * (1. - alpha) + alpha / nr_classes as f32;
+        }
+
+        let logits = logits.slice([0..batch_size, 0..1]).reshape([batch_size]);
```

Comment:
I'm not sure we need the `slice`, just the reshape should do the job; if the logits have more than just one element per batch, I think it's better to panic than to train with bugs.